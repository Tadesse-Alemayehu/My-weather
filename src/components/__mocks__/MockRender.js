import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../Redux/configureStore';

export default (component) => (
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>
);
