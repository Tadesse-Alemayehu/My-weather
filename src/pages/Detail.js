import React from 'react'
import { useSearchParams } from 'react-router-dom'
import { useSelector } from 'react-redux';
import { Container, Row, Col } from 'react-bootstrap'
export default function Detail() {
  const [getParam, setParam] = useSearchParams();
  let city = { id: getParam.get('id') }
  city = useSelector(store => {
    for (let thisCity of store.cities) {
      if (thisCity.id === city.id)
        return thisCity;
    }
  });
  return (
    city && <Container>
      <Row>
        <h1>{city.name}</h1>
        <h4>{city.weather[0].description}</h4>
        <small>max-tmp: {city.main.temp_max}</small>
        <small>min-temp{city.main.temp_min}</small>
      </Row>
      <Row ><Col>Cloud</Col> <Col >{city.clouds.all}%</Col></Row>
      <Row><Col>Wind</Col> <Col >
        speed: {city.wind.speed} |
        deg: {city.wind.deg}Â° |
        gust: {city.wind.gust} </Col></Row>
      <Row ><Col>Humidity</Col> <Col >{city.main.humidity}%</Col></Row>
      <Row ><Col>Sea level</Col> <Col >{city.main.sea_level}%</Col></Row>
      <Row ><Col>Population</Col> <Col >{city.population || 'Not available'}</Col></Row>
      <Row><Col>Approximate distance</Col> <Col >{city.distance && city.distance + "KM" || 'Not available'}</Col></Row>
    </Container> || <div>No city found</div>
  )
}
