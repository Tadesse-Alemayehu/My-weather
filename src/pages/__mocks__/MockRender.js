import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../Redux/configureStore';

const MockRender = (component) => (
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>
);

export default MockRender;
