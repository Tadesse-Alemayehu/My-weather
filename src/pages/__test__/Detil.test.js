
import Detail from '../Detail';
import '@testing-library/jest-dom';
import MockRender from '../__mocks__/MockRender'
import renderer from 'react-test-renderer'

describe('Snapshot check for the Detail page', () => {

  test('should match the snapshot', () => {
    const detailPageSnap = renderer
      .create(MockRender(<Detail />)).toJSON();
    expect(detailPageSnap).toMatchSnapshot();
  })
})