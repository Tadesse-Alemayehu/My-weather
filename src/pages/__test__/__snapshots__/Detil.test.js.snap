// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot check for the Detail page should match the snapshot 1`] = `
<div
  data-testid="city-detail-page"
  id="detail"
>
  <p
    className="warning"
  >
    No city found
  </p>
  <button
    className="home"
    onClick={[Function]}
  >
    Home
  </button>
</div>
`;
